src/Microsoft.AspNetCore.Rewrite/CodeRewriteExtensions.cs

```diff
@@ -35,69 +34,71 @@ public static RewriteOptions AddRules(this RewriteOptions options, List<Rule> ru
+            var builder = new UrlRewriteRuleBuilder();
+            var pattern = InputParser.ParseInputString(onMatch);
+
+            builder.AddUrlMatch(regex);
+            builder.AddUrlAction(pattern, actionType: ActionType.Rewrite, stopProcessing: stopProcessing);
+            options.Rules.Add(builder.Build());
             return options;
         }
 
-        /// <summary>
-        /// Redirect a path to another path.
-        /// </summary>
-        /// <param name="options">The Url rewrite options.</param>
-        /// <param name="regex">The string regex pattern to compare against the http context.</param>
-        /// <param name="newPath">The string to replace the path with (with capture parameters).</param>
-        /// <param name="stopRewriteOnSuccess">Whether or not to stop rewriting on success of rule.</param>
-        /// <returns></returns>
-        public static RewriteOptions RedirectPath(this RewriteOptions options, string regex, string newPath, bool stopRewriteOnSuccess = false)
+        public static RewriteOptions RedirectRule(this RewriteOptions options, string regex, string onMatch, int statusCode)
+        {
+            return RedirectRule(options, regex, onMatch, statusCode, stopProcessing: false);
+        }
+
+        public static RewriteOptions RedirectRule(this RewriteOptions options, string regex, string onMatch, int statusCode, bool stopProcessing)
         {
-            options.Rules.Add(new PathRule { MatchPattern = new Regex(regex, RegexOptions.Compiled, TimeSpan.FromMilliseconds(1)), OnMatch = newPath, OnCompletion = Transformation.Redirect });
+            var builder = new UrlRewriteRuleBuilder();
+            var pattern = InputParser.ParseInputString(onMatch);
+
+            builder.AddUrlMatch(regex);
+            builder.AddUrlAction(pattern, actionType: ActionType.Redirect, stopProcessing: stopProcessing);
+            options.Rules.Add(builder.Build());
             return options;
         }
 
-        /// <summary>
-        /// Redirect http to https.
-        /// </summary>
-        /// <param name="options">The Url rewrite options.</param>
-        /// <param name="sslPort">The port to redirect the scheme to.</param>
-        /// <returns></returns>
-        public static RewriteOptions RedirectScheme(this RewriteOptions options, int? sslPort)
+        public static RewriteOptions RedirectToHttps(this RewriteOptions options, int statusCode)
+        {
+            return RedirectToHttps(options, statusCode, stopProcessing: false);
+        }
+
+        public static RewriteOptions RedirectToHttps(this RewriteOptions options, int statusCode, bool stopProcessing)
         {
-            options.Rules.Add(new SchemeRule { SSLPort = sslPort, OnCompletion = Transformation.Redirect });
+            var builder = new UrlRewriteRuleBuilder();
+            var rulePattern = InputParser.ParseInputString("https://{HTTP_HOST}/{R:1}");
```

Comment:
When we get to profiling I bet your prior implementation was more efficient for this simple rule. But we can go back to it if we need to.